numpy/core/src/multiarray/dragon4.c

```diff
@@ -2229,20 +2245,20 @@ Dragon4_PrintFloat32(char *buffer, npy_uint32 bufferSize, npy_float32 value,
     }
 
     /* deconstruct the floating point value */
-    floatUnion.floatingPoint = value;
-    floatExponent = GetExponent_F32(&floatUnion);
-    floatMantissa = GetMantissa_F32(&floatUnion);
+    floatUnion.floatingPoint = *value;
+    floatMantissa = floatUnion.integer & bitmask_u32(23);
+    floatExponent = (floatUnion.integer >> 23) & bitmask_u32(8);
 
     /* output the sign */
-    if (IsNegative_F32(&floatUnion)) {
+    if ((floatUnion.integer >> 31) != 0) {
```

Comment:
For clarity, `bool floatSign = floatUnion.integer >> 31`  would match the above mantissa and exponent nicely